File: game-server/src/handlers/mission/get_mission_status.rs
use sr_proto::prost::Message;
use sr_proto::{GetMissionStatusCsReq, GetMissionStatusScRsp, Mission, MissionStatus};

pub async fn handle(req: &[u8]) -> GetMissionStatusScRsp {
    let req = GetMissionStatusCsReq::decode(req).unwrap_or_default();

    GetMissionStatusScRsp {
        finished_main_mission_id_list: req.main_mission_id_list,
        sub_mission_status_list: req
            .sub_mission_id_list
            .iter()
            .map(|id| Mission {
                id: *id,
                progress: 1,
                status: MissionStatus::MissionFinish.into(),
            })
            .collect(),
        ..Default::default()
    }
}

File: gameserver/handler/mission/GetMissionStatus.go
package mission

import (
	"github.com/yuvlian/qingque-sr/gameserver/session"
	"github.com/yuvlian/qingque-sr/pb"
	"github.com/yuvlian/qingque-sr/pb/cid"
)

func GetMissionStatus(s *session.Session) error {
	var req pb.GetMissionStatusCsReq
	if err := s.DecodeReq(&req); err != nil {
		return err
	}

	subMissions := make([]*pb.Mission, 0, len(req.SubMissionIdList))
	for _, id := range req.SubMissionIdList {
		subMissions = append(subMissions, &pb.Mission{
			Id:       id,
			Progress: 1,
			Status:   pb.MissionStatus_MISSION_FINISH,
		})
	}

	rsp := &pb.GetMissionStatusScRsp{
		FinishedMainMissionIdList: req.MainMissionIdList,
		SubMissionStatusList:      subMissions,
	}
	return s.Send(cid.GetMissionStatusScRsp, rsp)
}
